src/main/scala/za/co/absa/abris/avro/sql/CatalystDataToAvro.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.abris.avro.sql
+
+import java.security.InvalidParameterException
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.{GenericData, IndexedRecord}
+import org.apache.spark.sql.avro.AvroSerializer
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{Expression, UnaryExpression}
+import org.apache.spark.sql.types.{BinaryType, DataType}
+import za.co.absa.abris.avro.format.SparkAvroConversions
+import za.co.absa.abris.avro.parsing.utils.AvroSchemaUtils
+import za.co.absa.abris.avro.read.confluent.SchemaManager
+
+case class CatalystDataToAvro(
+   child: Expression,
+   schemaProvider: SchemaProvider,
+   schemaRegistryConf: Option[Map[String,String]],
+   prependSchemaId: Boolean)
```

Comment:
Same as in `AvroDataToCatalyst`, maybe renaming to `confluentCompliant` would be more extensible and also more informative of why this is here.